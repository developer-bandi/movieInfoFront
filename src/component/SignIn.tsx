import { Link } from 'react-router-dom';
import styled from 'styled-components';
import kakao5 from '../tempdata/kakaoicon.png';
interface SignInProps {
  setId: React.Dispatch<React.SetStateAction<string>>;
  setPassword: React.Dispatch<React.SetStateAction<string>>;
  checkLogin: () => Promise<void>;
}

const SignIn = ({ setId, setPassword, checkLogin }: SignInProps) => {
  return (
    <SignInBlock>
      <Title>로그인</Title>
      <section>
        <NaverLoginButton
          href={`${process.env.REACT_APP_SERVER}/auth/naver`}
          target="_blank"
        >
          <NaverContentBlock>
            <NaverLogo>N</NaverLogo>
            <NaverText>네이버 로그인</NaverText>
          </NaverContentBlock>
        </NaverLoginButton>
        <KakaoLoginButton
          href={`${process.env.REACT_APP_SERVER}/auth/kakao`}
          target="_blank"
        >
          <KakaoContentBlock>
            <KakaologoImg src={kakao5} alt="x"></KakaologoImg>
            <KakaoLoginText>카카오 로그인</KakaoLoginText>
          </KakaoContentBlock>
        </KakaoLoginButton>
        <Notice>
          네이버 로그인의 경우 검수를 받지않아 관리자만 이용할수 있습니다
        </Notice>
      </section>
      <ContentBarBlock>
        <ContentBar />
        <ContentBarText>또는</ContentBarText>
        <ContentBar />
      </ContentBarBlock>
      <section>
        <InfoInput
          onChange={(e) => {
            setId(e.target.value);
          }}
          placeholder="아이디"
        />

        <InfoInput
          onChange={(e) => {
            setPassword(e.target.value);
          }}
          type="password"
          placeholder="비밀번호"
        />
        <LoginButton onClick={checkLogin}>로그인하기</LoginButton>
        <LocalLoginToolBlock>
          <LocalLoginTool>
            <Link to={'/signup'}>회원가입</Link>
          </LocalLoginTool>
          <LocalLoginTool>아이디 찾기</LocalLoginTool>
          <LocalLoginTool>비밀번호 찾기</LocalLoginTool>
        </LocalLoginToolBlock>
      </section>
    </SignInBlock>
  );
};

const SignInBlock = styled.main`
  width: 500px;
  margin: auto;
  margin-top: 3rem;
`;

const Title = styled.h1`
  font-size: 35px;
  font-weight: bold;
  width: 100%;
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 2rem;
`;

const NaverLoginButton = styled.a`
  width: 500px;
  height: 65px;
  background: #03c75a;
  border-radius: 12px;
  display: flex;
  align-items: center;
  margin-bottom: 1rem;
  cursor: pointer;
`;

const NaverContentBlock = styled.div`
  display: flex;
  margin: auto;
`;

const NaverLogo = styled.div`
  color: white;
  font-weight: 900;
  font-size: 30px;
  display: flex;
  align-items: center;
  margin-right: 10px;
`;

const NaverText = styled.div`
  color: white;
  font-size: 20px;
  text-align: center;
  display: flex;
  align-items: center;
`;

const KakaoLoginButton = styled.a`
  width: 500px;
  height: 65px;
  background: #fee500;
  border-radius: 12px;
  display: flex;
  align-items: center;
  margin-bottom: 2rem;
  cursor: pointer;
`;

const KakaoContentBlock = styled.div`
  display: flex;
  margin: auto;
`;

const KakaologoImg = styled.img`
  width: 30px;
  height: 65px;
  object-fit: cover;
  margin-right: 5px;
`;

const KakaoLoginText = styled.div`
  color: #000000;
  font-size: 20px;
  text-align: center;
  display: flex;
  align-items: center;
`;

const Notice = styled.div`
  width: 500px;
  text-align: center;
  color: red;
  margin-bottom: 1rem;
`;

const ContentBarBlock = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 2rem;
`;
const ContentBar = styled.hr`
  flex: auto;
  border: none;
  height: 1px;
  background-color: #e3e3e3;
`;

const ContentBarText = styled.span`
  padding: 0 10px;
  color: #b5b5b5;
`;

const InfoInput = styled.input`
  height: 3rem;
  width: 500px;
  display: block;
  margin: auto;
  margin-bottom: 1rem;
  border: none;
  border-bottom: 1px solid black;
  font-size: 20px;
  padding: 0;
`;

const LoginButton = styled.button`
  height: 65px;
  width: 500px;
  display: block;
  margin: auto;
  margin-bottom: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #39373a;
  border-radius: 12px;
  border: none;
  color: white;
  &:hover {
    opacity: 0.8;
    cursor: pointer;
  }
  font-size: 20px;
  margin-top: 2rem;
`;

const LocalLoginToolBlock = styled.ul`
  display: flex;
  justify-content: center;
`;

const LocalLoginTool = styled.li`
  padding: 0 1rem 0 1rem;
  cursor: pointer;
  & + & {
    border-left: 1px solid black;
  }
`;

export default SignIn;
