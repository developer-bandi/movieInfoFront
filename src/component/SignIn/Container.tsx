import SignInPresentational from './Presentational';
import useSignIn from './Hook';

const SignInContainer = () => {
  const { checkLogin, setId, setPassword } = useSignIn();
  return (
    <SignInPresentational
      setId={setId}
      setPassword={setPassword}
      checkLogin={checkLogin}
    ></SignInPresentational>
  );
};

export default SignInContainer;
