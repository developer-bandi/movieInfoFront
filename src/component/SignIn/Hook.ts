import { useState } from 'react';
import { useNavigate } from 'react-router-dom';
import { axiosPostLogin } from '../../lib/api';

const useSignIn = () => {
  const [userid, setId] = useState('');
  const [password, setPassword] = useState('');
  const navigate = useNavigate();
  const checkLogin = async () => {
    if (userid !== '' && password !== '') {
      try {
        const userInfo = await axiosPostLogin({
          userid,
          password,
        });
        if (userInfo.status === 202) {
          alert(userInfo.data);
        } else {
          alert('로그인에 성공하였습니다!');
          navigate('/');
        }
      } catch (err) {
        alert('서버에 에러가 발생하였습니다');
      }
    }
  };
  return {
    checkLogin,
    setId,
    setPassword,
  };
};

export default useSignIn;
