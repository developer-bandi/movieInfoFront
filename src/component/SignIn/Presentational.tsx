import { Link } from 'react-router-dom';
import kakao5 from '../../imgs/kakaoicon.png';
import styles from './Style';

interface SignInProps {
  setId: React.Dispatch<React.SetStateAction<string>>;
  setPassword: React.Dispatch<React.SetStateAction<string>>;
  checkLogin: () => Promise<void>;
}

const SignIn = ({ setId, setPassword, checkLogin }: SignInProps) => {
  return (
    <styles.MainBlock>
      <styles.Title>로그인</styles.Title>
      <section>
        <styles.NaverLoginButton
          href={`${process.env.REACT_APP_SERVER}/auth/naver`}
          target="_blank"
        >
          <styles.NaverContentBlock>
            <styles.NaverLogo>N</styles.NaverLogo>
            <styles.NaverText>네이버 로그인</styles.NaverText>
          </styles.NaverContentBlock>
        </styles.NaverLoginButton>
        <styles.KakaoLoginButton
          href={`${process.env.REACT_APP_SERVER}/auth/kakao`}
          target="_blank"
        >
          <styles.KakaoContentBlock>
            <styles.KakaologoImg src={kakao5} alt="x"></styles.KakaologoImg>
            <styles.KakaoLoginText>카카오 로그인</styles.KakaoLoginText>
          </styles.KakaoContentBlock>
        </styles.KakaoLoginButton>
        <styles.Notice>
          네이버 로그인의 경우 검수를 받지않아 관리자만 이용할수 있습니다
        </styles.Notice>
      </section>
      <styles.BarBlock>
        <styles.Bar />
        <styles.BarText>또는</styles.BarText>
        <styles.Bar />
      </styles.BarBlock>
      <section>
        <styles.InfoInput
          onChange={(e) => {
            setId(e.target.value);
          }}
          placeholder="아이디"
        />
        <styles.InfoInput
          onChange={(e) => {
            setPassword(e.target.value);
          }}
          type="password"
          placeholder="비밀번호"
        />
        <styles.LoginButton onClick={checkLogin}>로그인하기</styles.LoginButton>
        <styles.LocalLoginToolBlock>
          <styles.LocalLoginTool>
            <Link to={'/signup'}>회원가입</Link>
          </styles.LocalLoginTool>
        </styles.LocalLoginToolBlock>
      </section>
    </styles.MainBlock>
  );
};

export default SignIn;
