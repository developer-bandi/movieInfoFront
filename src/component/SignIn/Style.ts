import styled from 'styled-components';

const MainBlock = styled.main`
  width: 500px;
  margin: auto;
  margin-top: 3rem;
`;

const Title = styled.h1`
  font-size: 35px;
  font-weight: bold;
  width: 100%;
  text-align: center;
  margin-top: 1rem;
  margin-bottom: 2rem;
`;

const NaverLoginButton = styled.a`
  width: 500px;
  height: 65px;
  background: #03c75a;
  border-radius: 12px;
  display: flex;
  align-items: center;
  margin-bottom: 1rem;
  cursor: pointer;
`;

const NaverContentBlock = styled.div`
  display: flex;
  margin: auto;
`;

const NaverLogo = styled.div`
  color: white;
  font-weight: 900;
  font-size: 30px;
  display: flex;
  align-items: center;
  margin-right: 10px;
`;

const NaverText = styled.div`
  color: white;
  font-size: 20px;
  text-align: center;
  display: flex;
  align-items: center;
`;

const KakaoLoginButton = styled.a`
  width: 500px;
  height: 65px;
  background: #fee500;
  border-radius: 12px;
  display: flex;
  align-items: center;
  margin-bottom: 2rem;
  cursor: pointer;
`;

const KakaoContentBlock = styled.div`
  display: flex;
  margin: auto;
`;

const KakaologoImg = styled.img`
  width: 30px;
  height: 65px;
  object-fit: cover;
  margin-right: 5px;
`;

const KakaoLoginText = styled.div`
  color: #000000;
  font-size: 20px;
  text-align: center;
  display: flex;
  align-items: center;
`;

const Notice = styled.div`
  width: 500px;
  text-align: center;
  color: red;
  margin-bottom: 1rem;
`;

const BarBlock = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 2rem;
`;
const Bar = styled.hr`
  flex: auto;
  border: none;
  height: 1px;
  background-color: #e3e3e3;
`;

const BarText = styled.span`
  padding: 0 10px;
  color: #b5b5b5;
`;

const InfoInput = styled.input`
  height: 3rem;
  width: 500px;
  display: block;
  margin: auto;
  margin-bottom: 1rem;
  border: none;
  border-bottom: 1px solid black;
  font-size: 20px;
  padding: 0;
`;

const LoginButton = styled.button`
  height: 65px;
  width: 500px;
  display: block;
  margin: auto;
  margin-bottom: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #39373a;
  border-radius: 12px;
  border: none;
  color: white;
  &:hover {
    opacity: 0.8;
    cursor: pointer;
  }
  font-size: 20px;
  margin-top: 2rem;
`;

const LocalLoginToolBlock = styled.div`
  display: flex;
  justify-content: center;
`;

const LocalLoginTool = styled.div`
  cursor: pointer;
`;

const styles = {
  MainBlock,
  Title,
  NaverContentBlock,
  NaverLoginButton,
  NaverLogo,
  NaverText,
  KakaoContentBlock,
  KakaoLoginButton,
  KakaologoImg,
  KakaoLoginText,
  Notice,
  BarBlock,
  Bar,
  BarText,
  InfoInput,
  LoginButton,
  LocalLoginToolBlock,
  LocalLoginTool,
};

export default styles;
