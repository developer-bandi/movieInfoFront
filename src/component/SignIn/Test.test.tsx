import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import axios from 'axios';
import { act } from 'react-dom/test-utils';
import { MemoryRouter } from 'react-router-dom';
import useSignIn from './Hook';
import SignIn from './Presentational';

describe('SignIn Presentational', () => {
  it('전체 화면 체크', () => {
    const setIdMock = jest.fn();
    const setPassword = jest.fn();
    const checkLogin = jest.fn();
    const utils = render(
      <SignIn
        setId={setIdMock}
        setPassword={setPassword}
        checkLogin={checkLogin}
      />,
      { wrapper: MemoryRouter }
    );
    expect(utils.container).toMatchSnapshot();
    fireEvent.change(screen.getByPlaceholderText('아이디'), {
      target: { value: 'new value' },
    });
    fireEvent.change(screen.getByPlaceholderText('비밀번호'), {
      target: { value: 'new value' },
    });
    fireEvent.click(screen.getByTestId('login'));
    expect(setIdMock).toBeCalledTimes(1);
    expect(setPassword).toBeCalledTimes(1);
    expect(checkLogin).toBeCalledTimes(1);
  });
});

jest.mock('axios');
const mockedAxios = axios as jest.Mocked<typeof axios>;

describe('SignIn Hook', () => {
  window.alert = jest.fn();
  afterEach(cleanup);
  describe('checkLogin함수 테스트', () => {
    it('에러 발생', async () => {
      const { result } = renderHook(() => useSignIn(), {
        wrapper: MemoryRouter,
      });

      mockedAxios.post.mockImplementation(() =>
        Promise.reject({ status: 500, data: '에러발생' })
      );
      act(() => {
        result.current.setId('testid');
        result.current.setPassword('testpassword');
      });
      await act(async () => {
        await result.current.checkLogin();
      });

      expect(window.alert).toBeCalledWith('서버에 에러가 발생하였습니다');
      expect(window.alert).toBeCalledTimes(1);
    });
    it('로그인 성공', async () => {
      const { result } = renderHook(() => useSignIn(), {
        wrapper: MemoryRouter,
      });
      mockedAxios.post.mockImplementation(() =>
        Promise.resolve({ status: 200, data: { userInfo: {} } })
      );
      act(() => {
        result.current.setId('testid');
        result.current.setPassword('testpassword');
      });
      await act(async () => {
        await result.current.checkLogin();
      });
      expect(window.alert).toBeCalledWith('로그인에 성공하였습니다!');
    });
    it('로그인 실패', async () => {
      const { result } = renderHook(() => useSignIn(), {
        wrapper: MemoryRouter,
      });
      mockedAxios.post.mockImplementation(() =>
        Promise.resolve({ status: 202, data: '로그인 실패' })
      );
      act(() => {
        result.current.setId('testid');
        result.current.setPassword('testpassword');
      });
      await act(async () => {
        await result.current.checkLogin();
      });
      expect(window.alert).toBeCalledWith('로그인 실패');
    });
  });
});
