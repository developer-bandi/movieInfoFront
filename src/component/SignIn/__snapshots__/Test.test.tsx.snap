// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Presentational 전체 화면 체크 1`] = `
<div>
  <main
    class="sc-bdvvtL oTaWz"
  >
    <h1
      class="sc-gsDKAQ cnDTFv"
    >
      로그인
    </h1>
    <section>
      <a
        class="sc-dkPtRN bBFgeL"
        href="http://localhost:8001/auth/naver"
        target="_blank"
      >
        <div
          class="sc-hKwDye kauDnZ"
        >
          <div
            class="sc-eCImPb fTNhwO"
          >
            N
          </div>
          <div
            class="sc-jRQBWg cUqyvw"
          >
            네이버 로그인
          </div>
        </div>
      </a>
      <a
        class="sc-gKclnd dLVobK"
        href="http://localhost:8001/auth/kakao"
        target="_blank"
      >
        <div
          class="sc-iCfMLu hcVoBx"
        >
          <img
            alt="x"
            class="sc-furwcr cjOFej"
            src="kakaoicon.png"
          />
          <div
            class="sc-pVTFL eHzQxr"
          >
            카카오 로그인
          </div>
        </div>
      </a>
      <div
        class="sc-jrQzAO kjDrFI"
      >
        네이버 로그인의 경우 검수를 받지않아 관리자만 이용할수 있습니다
      </div>
    </section>
    <div
      class="sc-kDTinF jDzdAA"
    >
      <hr
        class="sc-iqseJM XJnxr"
      />
      <span
        class="sc-crHmcD jhAqhz"
      >
        또는
      </span>
      <hr
        class="sc-iqseJM XJnxr"
      />
    </div>
    <section>
      <input
        class="sc-egiyK dzlrIo"
        placeholder="아이디"
      />
      <input
        class="sc-egiyK dzlrIo"
        placeholder="비밀번호"
        type="password"
      />
      <button
        class="sc-bqiRlB kqNncU"
        data-testid="login"
      >
        로그인하기
      </button>
      <div
        class="sc-ksdxgE jsSzDP"
      >
        <div
          class="sc-hBUSln eAfvkl"
        >
          <a
            href="/signup"
          >
            회원가입
          </a>
        </div>
      </div>
    </section>
  </main>
</div>
`;
